rfq/negotiator.go

```diff
@@ -157,66 +157,53 @@ func (n *Negotiator) HandleOutgoingBuyOrder(buyOrder BuyOrder) error {
 // queryAskFromPriceOracle queries the price oracle for an asking price. It
 // returns an appropriate outgoing response message which should be sent to the
 // peer.
-func (n *Negotiator) queryAskFromPriceOracle(
-	request rfqmsg.BuyRequest) (rfqmsg.OutgoingMsg, error) {
+func (n *Negotiator) queryAskFromPriceOracle(peer *route.Vertex,
+	assetId *asset.ID, assetGroupKey *btcec.PublicKey, assetAmount uint64,
+	bid *lnwire.MilliSatoshi) (*lnwire.MilliSatoshi, uint64, error) {
 
 	// Query the price oracle for an asking price.
 	ctx, cancel := n.WithCtxQuitNoTimeout()
 	defer cancel()
 
 	oracleResponse, err := n.cfg.PriceOracle.QueryAskPrice(
-		ctx, request.AssetID, request.AssetGroupKey,
-		request.AssetAmount, &request.BidPrice,
+		ctx, assetId, assetGroupKey, assetAmount, bid,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("failed to query price oracle for ask "+
-			"price: %w", err)
+		return nil, 0, fmt.Errorf("failed to query price oracle for "+
+			"ask price: %w", err)
 	}
 
-	// If the price oracle returned an error, then we will return a quote
-	// reject message which contains the error message supplied by the
-	// price oracle.
+	// Now we will check for an error in the response from the price oracle.
+	// If present, we will convert it to a string and return it as an error.
 	if oracleResponse.Err != nil {
-		rejectErr := rfqmsg.NewErrPriceOracleError(
-			oracleResponse.Err.Code, oracleResponse.Err.Msg,
-		)
-
-		reject := rfqmsg.NewReject(request, rejectErr)
-		return reject, nil
+		return nil, 0, fmt.Errorf("failed to query price oracle for "+
+			"ask price: %s", oracleResponse.Err)
 	}
 
-	// By this point the price oracle, did not return an error. However, if
-	// the asking price is nil, then we will return a quote reject
-	// message indicating that the price oracle did not specify an error.
+	// By this point, the price oracle did not return an error. However, if
+	// the asking price is nil, then we will return early.
 	if oracleResponse.AskPrice == nil {
-		rejectErr := rfqmsg.ErrPriceOracleUnspecifiedError
-
-		reject := rfqmsg.NewReject(request, rejectErr)
-		return reject, nil
+		return nil, 0, nil
```

Comment:
I think maybe the price should be a `fn.Option` then? Returning `nil` alongside a `nil` error is exactly the pattern we want to avoid at all cost I think.

Reply:
Or alternatively create a specific error type to catch at the call site.

Reply:
I don't think returning a price of `nil` or `fn.None` in combination of `err == nil` should ever be correct here. I think we need a `Result` type, see: https://github.com/lightninglabs/taproot-assets/issues/832 What do you think?

Reply:
I'm not sure I understand. If the combination of `fn.None` and `err == nil` should never be correct, why are we returning it then? Wouldn't the "never be correct" indicate it should actually be an error type?
Using the `Result` type is just a more formal contract of what we already implicitly want to do in Golang (never `return nil, nil`), just with new syntax, which IMO breaks a lot of existing visual pattern we already got used to over the years.

Reply:
> I'm not sure I understand. If the combination of fn.None and err == nil should never be correct, why are we returning it then? Wouldn't the "never be correct" indicate it should actually be an error type?

Ah, you're right, I wasn't thinking clearly on this. I'll return an error type instead of `return nil, 0, nil`.